test/functional/ui/output_spec.lua

```diff
@@ -22,6 +22,7 @@ describe("shell command :!", function()
 
   after_each(function()
     screen:detach()
+    session.command('call jobstop(b:terminal_job_id)')
```

Comment:
Well, screen tests are by nature indeterministic, nvim never says "drawing done, the screen will not change until new input". Basically the test passes if the correct state is seen, and no more updates are already on the update queue (well updates correct=>correct is allowed, but correct=>wrong=>correct is considered a failure). `screen:redraw_debug()` can be used to see all intermediate states.
BTW thanks for fixing my mistake from #5197.